tests/resources/aria-at-harness.mjs

```diff
@@ -0,0 +1,372 @@
+    window.setTimeout(() => {
+      displayTestPageAndInstructions(testPage);
+    }, 100);
+    return;
+  }
+
+  testPageWindow = window.open(testPage, '_blank', 'toolbar=0,location=0,menubar=0');
+
+  var style = document.createElement('style');
+  style.innerHTML = PAGE_STYLES;
+  document.head.appendChild(style);
+
+  if (allBehaviors.length > 0) {
+    displayInstructionsForBehaviorTest(0);
+  }
+}
+
+function displayInstructionsForBehaviorTest(behaviorId) {
+  // First, execute necesary set up script in test page
+  executeScriptInTestPage();
+
+  const totalBehaviors = allBehaviors.length;
+  const behavior = allBehaviors[behaviorId];
+  document.body.innerHTML = (TEST_HTML_OUTLINE);
+
+  const mode = behavior.mode[0];
+  const modeInstructions = getModeInstructions(mode, at);
+  const userInstructions = behavior.specific_user_instruction;
+  const commands = behavior.commands;
+  const assertions = behavior.assertions;
+
+  let instructionsEl = document.getElementById('instructions');
+  instructionsEl.innerHTML = `
+<h1 id="behavior-header" tabindex="0">Testing behavior ${behaviorId+1} of ${totalBehaviors}</h1>
+<h2>Instructions</h2>
+<ol>
+  <li><em>${modeInstructions}</em></li>
+  <li>Then, <em>${userInstructions}</em> using each of the following <emp>${at}<emp> controls:</li>
+  <ul id='at_controls' aria-label='AT controls'>
+  </ul>
+</ol>
+<h2>Success Criteria</h2>
+<p>For this test to pass, the following assertions must be met for every possible command:</p>
+<ul id='assertions'>
+</ul>
+`;
+
+  for (let command of commands) {
+    let el = document.createElement('li');
+    el.innerHTML = `<em>${command}</em>`;
+    document.getElementById('at_controls').append(el);
+  }
+
+  for (let assertion of assertions) {
+    let el = document.createElement('li');
+    el.innerHTML = `<em>${assertion}</em>`;
+    document.getElementById('assertions').append(el);
+  }
+
+  let recordResults = `<h2>Record Results</h2><p>${document.title}</p>`;
+
+  for (let c = 0; c < commands.length; c++) {
+    recordResults += `<table id="cmd-${c}" class="record-results">
+<tr>
+  <td>Results for command: ${commands[c]}</td>
+  <td>
+    <input type="checkbox" id="allpass-${c}" class="allpass" name="allpass">
+    <label for="allpass-${c}">All Pass</label>
+  </td>
+  <td>
+    <input type="checkbox" id="allfail-${c}" class="allfail" name="allfail">
+    <label for="allfaill-${c}">All Fail</label>
+  </td>
+  <td>Last Utterance</td>
+  <td>Other Details</td>
+</tr>
+`;
+
+    for (let a = 0; a < assertions.length; a++) {
+      recordResults += `
+<tr>
+  <td>${assertions[a]}</td>
+  <td>
+    <input type="checkbox" id="pass-${c}-${a}" class="pass" name="pass-${c}-${a}">
+    <label for="pass-${c}-${a}">Pass</label>
+  </td>
+  <td>
+    <input type="checkbox" id="fail-${c}-${a}" class="fail" name="fail-${c}-${a}">
+    <label for="fail-${c}-${a}">Fail</label>
+  </td>
+  <td>
+    <input type="text" id="utterance-${c}-${a}">
```

Comment:
As mentioned on the call, there are a few a11y issues here. Consider running the page through axe to help identify, fix, and validate. https://www.deque.com/axe/

Reply:
Great tip thanks axe was really helpful! I fixed all issues except for the accessible name for the radios.. I want to wait for Matt's feedback on what the accessible names should be.